spring-cloud-stream-binders/spring-cloud-stream-binder-kafka/src/main/java/org/springframework/cloud/stream/binder/kafka/KafkaMessageChannelBinder.java

```diff
@@ -429,28 +448,53 @@ protected void doStop() {
 	 * Creates a Kafka topic if needed, or try to increase its partition count to the
 	 * desired number.
 	 */
-	private Collection<Partition> ensureTopicCreated(final String topicName, final int minExpectedPartitions,
-			int replicationFactor) {
+	private Collection<Partition> ensureTopicCreated(final String topicName, final int partitionCount) {
 
 		final ZkClient zkClient = new ZkClient(zkAddress, getZkSessionTimeout(), getZkConnectionTimeout(),
 				ZKStringSerializer$.MODULE$);
 		try {
-			// The following is basically copy/paste from AdminUtils.createTopic() with
-			// createOrUpdateTopicPartitionAssignmentPathInZK(..., update=true)
 			final Properties topicConfig = new Properties();
-			if (isAutoConfigureTopics()) {
-				Seq<Object> brokerList = ZkUtils.getSortedBrokerList(zkClient);
-				final scala.collection.Map<Object, Seq<Object>> replicaAssignment = AdminUtils
-						.assignReplicasToBrokers(brokerList, minExpectedPartitions, replicationFactor, -1, -1);
-				metadataRetryOperations.execute(new RetryCallback<Object, RuntimeException>() {
-
-					@Override
-					public Object doWithRetry(RetryContext context) throws RuntimeException {
-						AdminUtils.createOrUpdateTopicPartitionAssignmentPathInZK(zkClient, topicName, replicaAssignment,
-								topicConfig, true);
-						return null;
+			TopicMetadata topicMetadata = AdminUtils.fetchTopicMetadataFromZk(topicName, zkClient);
+			if (topicMetadata.errorCode() == ErrorMapping.NoError()) {
+				// only consider minPartitionCount for resizing if autoAddPartitions is true
+				int effectivePartitionCount = isAutoAddPartitions() ? Math.max(defaultMinPartitionCount, partitionCount)
+						: partitionCount;
```

Comment:
It's not clear why `defaultMinPartitionCount` wins if it's larger than the requested partition count, but I see from the doc changes that this was your intention.

That said, the code in the configuration...
    		kafkaMessageChannelBinder.setDefaultMinPartitionCount(kafkaBinderConfigurationProperties.getMinPartitionCount());

... and the docs talking about `minPartitionCount`, tell me that this property should really be called `minPartitionCount` if it truly is an absolute minimum rather than a default minimum.